File: Array&String/HashTable/Python/find_the_town_judge.py
class Solution:
    def findJudge(self, n: int, trust: list[list[int]]) -> int:
        if not trust: return n if n == 1 else -1

        fr, to = set(), {}

        for a, b in trust:
            fr.add(a)

            if a in to:
                to.pop(a)

            if b not in fr:
                to[b] = to.get(b, 0) + 1

        if (len(to) > 1 or not to or len(fr) >= n):
            return -1
        else:
            judge = next(iter(to))
            if to[judge] == n-1: return judge
            else: return -1

    # shoreter
    def findJudge(self, n: int, trust: list[list[int]]) -> int:
        if not trust: return n if n == 1 else -1
        trust_count = [0] * (n + 1)
        for a, b in trust:
            trust_count[a] -= 1
            trust_count[b] += 1
        for i in range(1, n + 1):
            if trust_count[i] == n - 1:
                return i
        return -1

File: Array&String/HashTable/CPP/find_the_town_judge.cpp
#include <vector>
using namespace std;

class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {
        vector<int> town_judge(n+1, 0);
        for (vector<int> rel: trust) {
            town_judge[rel[0]]--;
            town_judge[rel[1]]++;
        }

        for (int i = 1; i < n+1; i++) {
            if (town_judge[i] == n-1) return i;
        }
        return -1;
    }
};